server/src/main/java/org/elasticsearch/cluster/routing/allocation/NodeAllocationStatsCalculator.java

```diff
@@ -24,15 +24,21 @@
 
 import java.util.Map;
 
-public class NodeAllocationStatsProvider {
+/**
+ * Calculates the balancer weights and usage stats for each node: see {@link NodeShardAllocationStats} for details.
+ */
+public class NodeAllocationStatsCalculator {
     private final WriteLoadForecaster writeLoadForecaster;
 
     private volatile float indexBalanceFactor;
     private volatile float shardBalanceFactor;
     private volatile float writeLoadBalanceFactor;
     private volatile float diskUsageBalanceFactor;
 
-    public record NodeAllocationAndClusterBalanceStats(
+    /**
+     * Node shard allocation stats, and the total node weight.
+     */
+    public record NodeShardAllocationStats(
```

Comment:
I'm not sure what is the benefit of this rename? The original seems more descriptive.

Reply:
I didn't see the connection to the `ClusterBalanceStats` part: the fields are node specific, there doesn't appear to be any cluster level information included. Could you elaborate how you were thinking about it?

Reply:
As mentioned [here](https://github.com/elastic/elasticsearch/pull/119613/files#r1907240191), only a subset of the stats is related to shard allocation, the other part is node weights. Just `NodeAllocationStats` was taken, so I've just spelled out both in the name. Maybe there is a more generic name, but `NodeShardAllocationStats` is misleading IMO. maybe `NodeAllocationStatsAndWeight`? I'm not sure.

Reply:
Updated to `NodeAllocationStatsAndWeight` 👍  I think it very clearly differentiates it from `NodeAllocationStats`, too, which is nice.